commandapi-core/src/main/java/dev/jorel/commandapi/arguments/WorldArgument.java

```diff
@@ -0,0 +1,75 @@
+package dev.jorel.commandapi.arguments;
+
+import com.mojang.brigadier.StringReader;
+import com.mojang.brigadier.arguments.StringArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import dev.jorel.commandapi.nms.NMS;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An argument that represents the Bukkit {@link World} object
+ *
+ * @apiNote Returns a {@link World} object
+ */
+public class WorldArgument extends Argument<World> {
+
+	public WorldArgument(String nodeName) {
+		super(nodeName, StringArgumentType.word());
+
+		addSuggestions();
+	}
+
+	private void addSuggestions() {
+		super.replaceSuggestions((info, builder) -> {
+			String currentArg = info.currentArg();
+
+			String[] worlds = new String[Bukkit.getWorlds().size()];
+			List<World> worldsList = Bukkit.getWorlds();
+			for (World world : worldsList) {
+				worlds[worldsList.indexOf(world)] = world.getName();
+			}
+
+			List<String> worldSuggestions = new ArrayList<>();
+			for (String world : worlds) {
+				if (world.startsWith(currentArg)) {
+					worldSuggestions.add(world);
+				}
+			}
+
+			for (String result : worldSuggestions) {
+				builder.suggest(result);
+			}
```

Comment:
This looks like a _really really really_ complicated way of doing:
```java
for (World world : Bukkit.getWorlds()) {
	final String worldName = world.getName();
	if(worldName.startsWith(currentArg)) {
		builder.suggest(worldName);
	}
}
```
Care to justify why you implemented this with three for loops?

Reply:
Also, I don't think you need to check if `worldName.startsWith(currentArg)`, because I think `builder#suggest` already does that (relating to [this comment in discord](https://discord.com/channels/745416925924032523/745418840153784401/1042923222393421904)).

Reply:
Yeah... no. After seeing your solution, I actually feel pretty dumb when looking at my solution...

Reply:
> Also, I don't think you need to check if `worldName.startsWith(currentArg)`, because I think `builder#suggest` already does that (relating to [this comment in discord](https://discord.com/channels/745416925924032523/745418840153784401/1042923222393421904)).

Actually, I tried it without that first and the suggestions didn't work at all.

Reply:
Oh, huh, looking at `builder#suggest`, here's what it has:
```java
 public SuggestionsBuilder suggest(String text) {
    if (text.equals(this.remaining)) {
        return this;
    } else {
        this.result.add(new Suggestion(StringRange.between(this.start, this.input.length()), text));
        return this;
    }
}
```
I think I was confusing that with [`ArgumentSuggestions#shouldSuggest`](https://github.com/JorelAli/CommandAPI/blob/dev/dev/commandapi-core/src/main/java/dev/jorel/commandapi/arguments/ArgumentSuggestions.java#L282) which dose filter out suggestions with `.startsWith`.